src/test/java/gov/healthit/chpl/aqa/asserts/ViewDeveloperPageAsserts.java

```diff
@@ -0,0 +1,154 @@
+    }
+
+    /**
+     * Assert that "Field is required" error message is displayed for Developer Name.
+     * @param expectedErrorMessage is the expected error message
+     */
+    @Then("^I see \"(.*)\" error message for developer name field$")
+    public void iSeeFieldIsRequiredErrorForDeveloperName(final String expectedErrorMessage) {
+        WebDriverWait wait = new WebDriverWait(getDriver(), LONG_TIMEOUT);
+        wait.until(ExpectedConditions.visibilityOf(ViewDeveloperPage.developerNameMissingErrorMessage(getDriver())));
+        String actualErrorMessage = ViewDeveloperPage.developerNameMissingErrorMessage(getDriver()).getText();
+        assertEquals(actualErrorMessage, expectedErrorMessage);
+    }
+
+    /**
+     * Assert that "Field is required" error message is displayed for Developer Name.
+     * @param expectedErrorMessage is the expected error message
+     */
+    @Then("^I see \"(.*)\" error message for email field$")
+    public void iSeeFieldIsRequiredErrorForEmail(final String expectedErrorMessage) {
+        WebDriverWait wait = new WebDriverWait(getDriver(), LONG_TIMEOUT);
+        wait.until(ExpectedConditions.visibilityOf(ViewDeveloperPage.emailMissingErrorMessage(getDriver())));
+        String actualErrorMessage = ViewDeveloperPage.emailMissingErrorMessage(getDriver()).getText();
+        assertEquals(actualErrorMessage, expectedErrorMessage);
+    }
+
+    /**
+     * Assert that expected text is displayed in developers to merge section.
+     * @param expectedText is "At least one other Developer must be selected to merge"
+     */
+    @Then("^I see \"([^\"]*)\" in developers to merge section$")
+    public void iSeeMessageInDevelopersToMergeSection(final String expectedText) {
+        String actualText = ViewDeveloperPage.developerToMergeMessage(getDriver()).getText();
+        assertEquals(expectedText, actualText);
+    }
+
+    /**
+     * Assert that expected text is displayed in Products moving to new Developer section.
+     * @param expectedText is "At least one Product must be selected to move"
+     */
+    @Then("^I see \"([^\"]*)\" in products moving to new developer section$")
+    public void iSeeMessageInProductsMovingToNewDeveloperSection(final String expectedText) {
+        String actualText = ViewDeveloperPage.productsMovingToNewDeveloperMessage(getDriver()).getText();
+        assertEquals(actualText, expectedText);
+    }
+
+    /**
+     * Assert that developer edit link does not exist on view developer page.
+     * @param developerId is the developer id passed in page objects
+     */
+    @Then("^I do not see developer edit link for developer \"([^\"]*)\"$")
+    public void iDoNotSeeDeveloperEditLink(final String developerId) {
+        boolean developerEditLinkFound = false;
+        try {
+            developerEditLinkFound = ViewDeveloperPage.editDeveloperLink(getDriver(), developerId).isDisplayed();
+        } catch (NoSuchElementException e) {
+        }
```

Comment:
In this assert, you're expecting an exception to be thrown, right? If that's the case, I'd put the assert inside the catch block. Empty catch blocks are a code smell

Reply:
Okay, so if I'd put the assert inside the catch block the test case would never fail. Assert needs to be outside the catch block. Currently, after an exception is been thrown it executes the assert statement. 

Consider the scenario like below:-
Given I'm logged in as "ROLE_ONC"
When I navigate to the developer page for developer "1854"
Then I do not see developer edit link for developer "1854"

The test case fails because if I login - I'm able to see the developer edit link but if I have the assert statement inside the catch block it does not fail.

So instead of leaving the catch block empty I could print something like:- 
catch (NoSuchElementException e) {
            System.out.println("Developer split link is not available");
        }

Reply:
Please let me know if I'm correct

Reply:
In the try block, you're trying to find an element. Is that element found? If it is, then in that try block you should have something like `fail("Found element when shouldn't have")`. If it's not found, does it throw an exception? If it does, and that's what we _want_ to have happen, then you should catch that exception, and in the catch block put your `assertTrue(!developerEditLinkFound, "Error, developer edit link is available);`.

If not finding an element doesn't throw an exception, then there shouldn't be a try/catch block at all